### PR TITLE
VB-2562 - validation updates around session capacity and updated VSiPValidationException to include multiple error messages.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Adds validation updates around session capacity and allows multiple validation errors.
## What is the intent behind these changes?

Admin API improvements

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt [hunk 1]
```diff
@@ -262,3 +262,18 @@ data class ErrorResponse(
   ) :
     this(status.value(), errorCode, userMessage, developerMessage)
 }
+
+data class ValidationErrorResponse(
+  override val status: Int,
+  override val errorCode: Int? = null,
+  override val userMessage: String? = null,
+  val developerMessages: List<String>,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt:270]: should this field be renamed to say moreInfo?

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt:270]: see comment below

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt [hunk 2]
```diff
@@ -262,3 +262,18 @@ data class ErrorResponse(
...
+) : ErrorResponse(status, errorCode, userMessage, developerMessages.joinToString()) {
+  constructor(
+    status: HttpStatus,
+    errorCode: Int? = null,
+    userMessage: String? = null,
+    developerMessages: List<String>,
+  ) :
+    this(status.value(), errorCode, userMessage, developerMessages)
+}
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt:279]: > You could I suppose put a code per validation message. ` val validationItemMessages: List<ValidationItemMessage>,`
> 
> ```
> data class ValidationItemMessage(
>   val code: String,
>   val message: String,
> ) 
> ```

But the front end might need the other parameters to decide if the message needs to be retried. Also we would need the userMessage on staff UI.

[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt:279]: But the front end might need the other parameters to decide if the message needs to be retried. Also we would need the userMessage on staff UI.

[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt]: Spoke to Tom and updated PR

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**